umbraco/Umbraco-CMS
# New backoffice: examine management controller

# Notes
Beware that the serialization of Dictionary<string, object> does not work for System.Text.Json in .NET 6, this has however been fixed in .NET 7, see more on this issue: https://github.com/dotnet/runtime/issues/67588
## What does this mean for us though?
Well this means that most of our actions just straight-up throw exceptions for now. 
When testing, the Dictionary<string, object> should be commented out from the ViewModel, so we can assert at least the rest of the logic works as intended 👍 
## Other changes
- Added `ExamineManagerService` to handle logic for validating indexes, searchers & populators instead of that logic being in the controller. Could argue for a better service name, but can't think of a better name 😅 
- Created the `ExamineIndexViewModel`
- Created an `ExamineIndexViewModelFactory` to handle logic of creating the view model.

# How to test
- Comment out the `public IReadOnlyDictionary<string, object?> ProviderProperties { get; set; } = null!;` from ExamineIndexViewModel.cs
- Try and build the project, you will get another error in the ExamineIndexViewModel factory as we now are trying to map something that no longer exists, comment out line 42 aswell and you're good to go!
- Test the different endpoints using this postman collection and assert they do as described: https://www.dropbox.com/s/76zkugv38f2lc8x/ExamineManagement.postman_collection.json?dl=0
